core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3189,7 +3189,7 @@ protected Scope visitSubqueryExpression(SubqueryExpression node, Optional<Scope>
         @Override
         protected Scope visitSetOperation(SetOperation node, Optional<Scope> scope)
         {
-            checkState(node.getRelations().size() >= 2);
+            checkState(node.getRelations().size() == 2, "relations size must be 2");
```

Comment:
I don't think this is true. Union and Intersect can have multiple branches.

Reply:
AstBuilder always accepts 2 relations for union and intersect operations. Am I missing something? 

https://github.com/trinodb/trino/blob/aab8121c767c75e20b1d6d0a8a382a5e712f5990/core/trino-parser/src/main/java/io/trino/sql/parser/AstBuilder.java#L1369-L1370

Reply:
Yes, that's correct. It's how the AstBuilder constructs it. But the AST itself allows for any number of branches.

Reply:
I’m not quite sure what the expected next step is here. I reverted the change and added the following line to the corresponding block:
```java
checkState(relations.size() == 2, "CORRESPONDING requires 2 relations");
```

Reply:
Added another commit to ensure 2 relations in `Union` and `Intersect` classes.